components/List/examples/supplemental/ManualLayoutListItemCell.m

```diff
@@ -0,0 +1,355 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "ManualLayoutListItemCell.h"
+
+#import <MDFInternationalization/MDFInternationalization.h>
+
+#import "MaterialInk.h"
+#import "MaterialTypography.h"
+#import "MaterialTypographyScheme.h"
+
+static const CGFloat kDefaultMarginTop = 12.0;
+static const CGFloat kDefaultMarginBottom = 12.0;
+static const CGFloat kDefaultMarginLeading = 12.0;
+static const CGFloat kDefaultMarginTrailing = 12.0;
+static const CGFloat kDefaultVerticalLabelPadding = 6.0;
+
+@interface ManualLayoutListItemCell ()
+
+@property (nonatomic, assign) CGPoint lastTouch;
+@property (strong, nonatomic, nonnull) MDCInkView *inkView;
+
+@property (strong, nonatomic, nonnull) UIView *textContainer;
+@property (strong, nonatomic, nonnull) UILabel *overlineLabel;
+@property (strong, nonatomic, nonnull) UILabel *titleLabel;
+@property (strong, nonatomic, nonnull) UILabel *detailLabel;
+
+@end
+
+@implementation ManualLayoutListItemCell
+@synthesize mdc_adjustsFontForContentSizeCategory = _mdc_adjustsFontForContentSizeCategory;
+
+#pragma mark Object Lifecycle
+
+-(instancetype)initWithFrame:(CGRect)frame {
+  self = [super initWithFrame:frame];
+  if (self) {
+    [self commonInit];
```

Comment:
note: This name will collide in framework code.  Embed class name in method name, commonManualLayoutListItemCellInit